Test.cpp

```diff
@@ -63,11 +70,55 @@
 #define LLVM_VERSION_MAJOR 0
 #endif
 
+#ifndef DEFAULT_COMPILER_DIR
+#define DEFAULT_COMPILER_DIR "/usr/bin"
+#endif
+
+#ifndef DEFAULT_COMPILER_NAME
+#define DEFAULT_COMPILER_NAME "gcc"
+#endif
+
+#ifndef LLVM_FOUND
+#define LLVM_FOUND 0
+#endif
+
 // someone should provide the signature of the function now versioning
 // in the form of function pointer type.
-typedef int (*signature_t)(int);
+typedef float (*signature_t)(int);
+typedef int (*signature_t_new)(float);
+int ret_value = 0;
+
+template <typename T>
+void checkResult(T result, T expected){
+  static_assert(std::is_arithmetic<T>::value, "checkResult only supports numeric types");
+  if (std::fabs(result - expected) < 1e-5) {
+    std::cout << "PASSED" << std::endl;
+  }else{
+    std::cout << "FAILED: expected = " << expected << ", got = " << result << std::endl;
+    if(!ret_value)
+      ret_value=1;
+  }
+}
+
+int check_log_for_warning(const std::string &log_file, const std::string &expected) {
```

Comment:
perhaps better call it
```suggestion
int check_log_for_content(const std::string &log_file, const std::string &expected) {
```